src/shared/ui/utils/assert-unreachable.ts

```diff
@@ -0,0 +1,3 @@
+export function assertUnreachable(x: never): never {
+  throw new Error(`Incorrect BaseHash props.type value : ${x}`);
```

Comment:
Why this message is put here, in `shared/ui/utils`?

I'd suggest just inlining it in the component itself:
 
```ts
const x: never = props.type
throw new Error(`unexpected props.type: ${String(x)}`)
```